calculatorFrame.java

```diff
@@ -111,191 +104,192 @@ public void init() {
 		clearButtonPanel.add(clearAll);
 		content.add(clearButtonPanel,BorderLayout.NORTH);
 	}
-	
+
 	public void actionPerformed(ActionEvent e) {
-		
+
 		//One button
-		if(e.getSource() == one){
-			if (flag == false){
+		if(e.getSource() == one) {
```

Comment:
Actually, forget the making of a `JNumberButton` class, and check out http://docs.oracle.com/javase/7/docs/api/javax/swing/AbstractButton.html#getText()

It looks like you can just do something like

```java
JNumberButton numberButton;
if(numberButton = isNumberButton(e.getSource()) {
  System.out.println(firstNumString);
  firstNumString += numberButton.getText();
  System.out.println(firstNumString);
  textArea.setText(textArea.getText() + e.getActionCommand());
}
```